java/connector-node/risingwave-sink-jdbc/src/main/java/com/risingwave/connector/SnowflakeJDBCSinkConfig.java

```diff
@@ -0,0 +1,242 @@
+            throw new SQLException("Failed to configure authentication: " + e.getMessage(), e);
+        }
+
+        var conn = DriverManager.getConnection(jdbcUrl, props);
+        conn.setAutoCommit(isAutoCommit());
+        conn.setTransactionIsolation(Connection.TRANSACTION_READ_COMMITTED);
+        return conn;
+    }
+
+    /**
+     * Handles Snowflake-specific authentication by processing the auth.method property and setting
+     * up appropriate authentication properties. This is the main entry point for Snowflake
+     * authentication configuration.
+     *
+     * <p>Supported authentication methods: - password: Standard username/password authentication -
+     * key_pair_file: Key-pair authentication using a private key file path - key_pair_object:
+     * Key-pair authentication using PEM content (converted to PrivateKey object)
+     *
+     * @param props Properties to configure (will be modified in-place). Expected properties:
+     *     auth.method, password, private_key_file, private_key_pem, private_key_file_pwd
+     * @throws IOException if PEM parsing fails
+     * @throws GeneralSecurityException if key conversion fails
+     * @throws OperatorCreationException if decryption fails
+     * @throws PKCSException if PKCS decryption fails
+     * @throws SQLException if authentication setup fails
+     */
+    public static void handleSnowflakeAuth(Properties props)
+            throws IOException,
+                    GeneralSecurityException,
+                    OperatorCreationException,
+                    PKCSException,
+                    SQLException {
+        String authMethod = props.getProperty(PROP_AUTH_METHOD);
+        if (authMethod == null) {
+            // No auth method specified, use default password authentication
+            return;
+        }
+
+        props.remove(PROP_AUTH_METHOD);
+
+        if ("password".equalsIgnoreCase(authMethod)) {
+            // Password authentication - no additional processing needed
+            // The password property is already set
+        } else if ("key_pair_file".equalsIgnoreCase(authMethod)) {
+            // File-based key-pair authentication
+            // The private_key_file and optional private_key_file_pwd properties are already set
+            props.put("authenticator", "snowflake_jwt");
+        } else if (AUTH_METHOD_KEY_PAIR_OBJECT.equalsIgnoreCase(authMethod)) {
+            // Object-based key-pair authentication - convert PEM to PrivateKey object
+            String pem = props.getProperty(PROP_PRIVATE_KEY_PEM);
+            String passphrase = props.getProperty(PROP_PRIVATE_KEY_PWD);
+
+            PrivateKey privateKey = loadPrivateKeyFromPem(pem, passphrase);
+            props.put(PROP_PRIVATE_KEY, privateKey);
+            props.remove(PROP_PRIVATE_KEY_PEM);
+            props.remove(PROP_PRIVATE_KEY_PWD);
+            props.put("authenticator", "snowflake_jwt");
+            LOG.info("Loaded private key for Snowflake authentication");
```

Comment:
nits: `handleSnowflakeAuth` is expected to called very frequently. How about using debug or trace log here?